dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,275 @@
+
+        combined_results = []
+        for post in post_details:
+            category = category_dict.get(post["post_id"], {"category_id": None, "category_name": None})
+            post["category_id"] = category["category_id"]
+            combined_results.append(post)
+
+        return combined_results
+
+    def fetch_raw_data(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch and combine post details and categories.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of combined dictionaries containing post details and categories.
+        """
+        if created_at and comparison:
+            post_details = self.fetch_post_details(created_at, comparison)
+        else:
+            post_details = self.fetch_post_details()
+
+        post_ids = [post["post_id"] for post in post_details]
+        post_categories = self.fetch_post_categories(post_ids)
+        return self.combine_posts_with_categories(
+            post_details, post_categories
+        )
+
+    def extract(self, period: datetime, recompute: bool = False) -> list:
+        """
+        Extract data based on the period and recompute flag.
+
+        :param period: The datetime period to filter posts.
+        :param recompute: Flag to indicate if recompute is needed.
+        :return: List of combined dictionaries containing post details and categories.
+        """
+        data = []
+        if recompute:
+            data = self.fetch_raw_data()
+        else:
+            latest_activity = self.rawmemberactivities_collection.find_one(
+                sort=[("date", -1)]
+            )
+            latest_activity_date = latest_activity["date"] if latest_activity else None
+            print(f"Latest activity date: {latest_activity_date}")
+            if latest_activity_date is not None:
+                # Convert latest_activity_date string to datetime
+                latest_activity_date_datetime = self.converter.from_iso_format(latest_activity_date) if latest_activity_date else None
+                # Convert latest_activity_date_datetime to ISO format with milliseconds
+                latest_activity_date_iso_format = self.converter.to_iso_format(latest_activity_date_datetime) if latest_activity_date_datetime else None
+                period_iso_format = self.converter.to_iso_format(period)
+                print(f"Period ISO format: {period_iso_format}")
+                if latest_activity_date_iso_format >= period_iso_format:
+                    data = self.fetch_raw_data(latest_activity_date_iso_format, "gt")
+                else:
+                    data = self.fetch_raw_data(period_iso_format, "gte")
+                    print("data fetched: \n", data)
```

Comment:
Please remove the print statements. The testing of each module and multiple modules together should happen using unit and integration tests